autogen/src/gen_gql_schema.rs

```diff
@@ -612,39 +691,55 @@ fn write_all_input_types(
         .collect::<HashSet<String>>();
 
     let mut types_added = HashSet::new();
+
+    let mut write_fields = WriteFields {
+        writer,
+        name: String::new(),
+        schema: SchemaObject::default(),
+        defs: &defs,
+        extra_it: &mut extra_it,
+        scalars: &mut scalars,
+        types_added: &mut types_added,
+        written_directives: &mut HashSet::new(),
+        arr_valid: &mut ArrayValidation::default(),
+        obj_valid: &mut ObjectValidation::default(),
+    };
+
     for (name, input_type) in defs.iter() {
         let mut name = name.clone();
         first_char_to_upper(&mut name);
-        write_input_type(
-            writer,
-            name,
-            input_type.clone().into_object(),
-            &defs,
-            &mut scalars,
-            &mut extra_it,
-            &mut types_added,
-        )?;
+        write_fields.name = name.clone();
+        write_fields.schema = input_type.clone().into_object();
+        write_input_type(&mut write_fields)?;
     }
 
     let mut new_extra_it = BTreeMap::new();
 
     for (name, extra_type) in extra_it.into_iter() {
+        let mut write_fields = WriteFields {
+            writer,
+            name: name.clone(),
+            schema: SchemaObject::default(),
+            defs: &defs,
+            extra_it: &mut new_extra_it,
+            scalars: &mut scalars,
+            types_added: &mut types_added,
+            written_directives: &mut HashSet::new(),
+            arr_valid: &mut ArrayValidation::default(),
+            obj_valid: &mut ObjectValidation::default(),
+        };
```

Comment:
having this all over the file looks odd and repetitive, it's easier to mistake.

Have you considered all the function that reuse this under single struct impl and store the shared data in the instance itself instead of passing it to every function?